### PR TITLE
Refactor instances -> elements and improved examples and docs

### PR DESCRIPTION
**Pull request by DaJansenGit**
This brings some changes which would be better placed in seperate branches but where needed in this for easier development, I'm sorry for the mess:
- some improvements regarding examples and documentation in form of a template plugin that allows to give an example without the usage of any "real" plugins
- the refactoring of `instances`  --> `elements` as this is confusing for the users as we also named the classes elements.
- split up the `PrepareBPS` task which was not really specific to one goal into the tasks `FilterTZ` and `ProcessSlabsRoofs`.
- bump python to 3.10 (solve problems with geomeppy and eppy compability with 3.10, see #602 which I mergend into this branch for testing purposes)
- make sure that docs are build in CI even if tests fail
-

### CODE REVIEW

**Comment by DaJansenGit** [comment]
@Cudok I fixed the problem with the tests. This is ready for review.

File: bim2sim/examples/e1_template_plugin.py
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply bim2sim/examples/e1_template_plugin.py:1]: solved

File: docs/source/user-guide.md
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply docs/source/user-guide.md:1]: solved

File: docs/source/user-guide/Base-Plugin.md
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply docs/source/user-guide/Base-Plugin.md:1]: solved

**Review by Cudok — APPROVED** [review]

**Pull request merged**